src/main/java/org/liliya/hotelapp/ui/Console.java

```diff
@@ -0,0 +1,119 @@
+
+        }
+    }
+
+    private void registerApartment() {
+        System.out.print("Enter price: ");
+        double price = scanner.nextDouble();
+
+        apartmentService.register(price);
+        System.out.println("\nApartment successfully registered");
+    }
+
+
+    private void reserveApartment() throws ReservationException {
+        System.out.print("Enter client name: ");
+        String clientName = scanner.nextLine();
+
+        Client client = new Client(clientName);
+        boolean reserved = apartmentService.reserve(client);
+        if (!reserved) {
+            throw new ReservationException("No available apartments found.");
+        }
+
+        System.out.println("\nApartment reserved successfully for " + client);
+    }
+
+    private void releaseApartment() throws ReservationException {
+        System.out.print("Enter apartment's id: ");
+        int id = scanner.nextInt();
+        boolean released = apartmentService.release(id);
+        if (!released) {
+            throw new ReservationException("No apartments found with such id.");
+        }
+        System.out.println("\nApartment successfully released");
+    }
+
+    private void getPaginatedAndSortedApartments() {
+        System.out.print("Enter page number: ");
+        int page = scanner.nextInt();
+
+        System.out.print("Enter page size: ");
+        int size = scanner.nextInt();
+        scanner.nextLine();
+
+        System.out.print("Sort by (id, price, status, name): ");
+        String sortBy = scanner.nextLine();
+
+        Comparator<Apartment> comparator = getComparator(sortBy);
+        if (comparator == null) {
+            System.out.println("Invalid sort field.");
+            return;
+        }
+
+        List<Apartment> apartments = apartmentService.getPaginatedAndSortedApartments(page, size, comparator);
+        System.out.println("\n--- Paginated and Sorted Apartments ---");
+        apartments.forEach(System.out::println);
+    }
+
+    private Comparator<Apartment> getComparator(String sortBy) {
+        Comparator<Apartment> comparator;
+
+        switch (sortBy.toLowerCase()) {
+            case "id" -> comparator = Comparator.comparing(Apartment::getId);
+            case "price" -> comparator = Comparator.comparing(Apartment::getPrice);
+            case "status" -> comparator = Comparator.comparing(Apartment::getReservationStatus);
+            case "name" -> comparator = Comparator.comparing(apartment -> apartment.getClient().getName());
+            default -> comparator = Comparator.comparing(Apartment::getId);
+        }
+        return comparator;
+    }
+
+    private void showMenu() {
+        System.out.println("\n--- Hotel application ---\n");
+        System.out.println("1. Register apartment");
+        System.out.println("2. Reserve an apartment");
+        System.out.println("3. Release an apartment");
+        System.out.println("4. Get apartments");
+        System.out.print("Select an option: ");
```

Comment:
could be text block https://www.baeldung.com/java-text-blocks

Reply:
fixed